### PR TITLE
fix: cases where bad incoming blocks could bypass a ban

### PR DESCRIPTION
**Pull request by SWvheerden**
Description
---
In some cases incoming blocks can escape a ban. 

Motivation and Context
---
Participating in bad behavior should result in a network ban.

### CODE REVIEW

**Review by stringhandler — COMMENTED** [review]
Not really a fan of this approach, it would be better to ban higher up. I don't know if it's possible, but this way introduces a lot of duplicate code

File: base_layer/core/src/base_node/comms_interface/inbound_handlers.rs [hunk 1]
```diff
@@ -506,7 +503,35 @@ where B: BlockchainBackend + 'static
...
+    /// Handles a `NewBlock` message. Only a single `NewBlock` message can be handled at once to prevent extraneous
+    /// requests for the full block.
+    /// This may (asynchronously) block until the other request(s) complete or time out and so should typically be
+    /// executed in a dedicated task.
+    pub async fn handle_new_block_message(
+        &mut self,
+        new_block: NewBlock,
+        source_peer: NodeId,
+    ) -> Result<(), CommsInterfaceError> {
+        if let Err(err @ CommsInterfaceError::ChainStorageError(ChainStorageError::ValidationError { .. })) =
```
Comment (stringhandler) [inline base_layer/core/src/base_node/comms_interface/inbound_handlers.rs]: You are swallowing all the other error types here.... I think this is worse

File: base_layer/core/src/base_node/comms_interface/inbound_handlers.rs [hunk 2]
```diff
@@ -520,13 +545,21 @@ where B: BlockchainBackend + 'static
                     self.blockchain_db.get_chain_metadata().await?.height_of_longest_chain(),
                 )
                 .await?;
-            return Err(CommsInterfaceError::InvalidBlockHeader(
-                BlockHeaderValidationError::ProofOfWorkError(PowError::AchievedDifficultyBelowMin),
-            ));
+            let err = BlockHeaderValidationError::ProofOfWorkError(PowError::AchievedDifficultyBelowMin);
```
Comment (stringhandler) [inline base_layer/core/src/base_node/comms_interface/inbound_handlers.rs:552]: Why not let this be caught higher up?

File: base_layer/core/src/base_node/comms_interface/inbound_handlers.rs (hunk 1 above)
Comment (SWvheerden) [reply base_layer/core/src/base_node/comms_interface/inbound_handlers.rs]: your right,this is worse

**Comment by SWvheerden** [comment]
closing this in favour of #5693

**Pull request closed without merging**